test/Subaccount/ClientTest.php

```diff
@@ -0,0 +1,369 @@
+<?php
+
+declare(strict_types=1);
+
+namespace VonageTest\Subaccount;
+
+use Laminas\Diactoros\Response;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Http\Message\RequestInterface;
+use Vonage\Client;
+use Vonage\Client\APIResource;
+use Vonage\Subaccount\Client as SubaccountClient;
+use Vonage\Subaccount\Filter\Subaccount;
+use Vonage\Subaccount\Request\NumberTransferRequest;
+use Vonage\Subaccount\Request\TransferRequest;
+use Vonage\Subaccount\SubaccountObjects\Account;
+use Vonage\Subaccount\SubaccountObjects\BalanceTransfer;
+use Vonage\Subaccount\SubaccountObjects\CreditTransfer;
+use VonageTest\Psr7AssertionTrait;
+use VonageTest\VonageTestCase;
+
+class ClientTest extends VonageTestCase
+{
+    use Psr7AssertionTrait;
+
+    protected APIResource $api;
+
+    protected Client|ObjectProphecy $vonageClient;
+    protected $subaccountClient;
+
+    public function setUp(): void
+    {
+        $this->vonageClient = $this->prophesize(Client::class);
+        $this->vonageClient->getCredentials()->willReturn(
+            new Client\Credentials\Basic('abc', 'def'),
+        );
+
+        /** @noinspection PhpParamsInspection */
+        $this->api = (new APIResource())
+            ->setIsHAL(true)
+            ->setErrorsOn200(false)
+            ->setClient($this->vonageClient->reveal())
+            ->setBaseUrl('https://api.nexmo.com/accounts');
+
+        $this->subaccountClient = new SubaccountClient($this->api);
+    }
+
+    public function testClientInitialises(): void
+    {
+        $this->assertInstanceOf(SubaccountClient::class, $this->subaccountClient);
+    }
+
+    public function testUsesCorrectAuth(): void
+    {
+        $this->vonageClient->send(Argument::that(function (RequestInterface $request) {
+            $this->assertEquals(
+                'Basic ',
+                mb_substr($request->getHeaders()['Authorization'][0], 0, 6)
+            );
+            return true;
+        }))->willReturn($this->getResponse('get-success'));
+
+        $apiKey = 'something';
+        $response = $this->subaccountClient->getPrimaryAccount($apiKey);
+    }
+
+    public function testWillPatchSubaccount(): void
```

Comment:
I know it's using a PATCH method, but from a "what are we doing" perspective we are updating the subaccount

```suggestion
    public function testWillUpdateSubaccount(): void
```